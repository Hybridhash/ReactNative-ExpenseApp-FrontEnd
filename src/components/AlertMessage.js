import React from 'react';
import {View, Modal, Text, StyleSheet, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';

const AlertMessage = ({message, visible, onClose}) => {
  return (
    <Modal
      animationType="fade"
      transparent={true}
      visible={visible}
      onRequestClose={onClose}
    >
      <View style={styles.container}>
        <View style={styles.innerContainer}>
          {message.map((msg, index) => (
            <Text key={index} style={styles.message}>{msg}</Text>
          ))}
          <TouchableOpacity onPress={onClose}>
            <Text style={styles.button}>Close</Text>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  );
};

AlertMessage.propTypes = {
  message: PropTypes.array.isRequired,
  visible: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.5)',
  },
  innerContainer: {
    backgroundColor: '#fff',
    padding: 20,
    borderRadius: 5,
  },
  message: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  button: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#4455BB',
    textAlign: 'center',
  },
});

export default AlertMessage;
