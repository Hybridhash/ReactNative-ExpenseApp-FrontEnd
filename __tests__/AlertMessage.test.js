import React from 'react';
import renderer from 'react-test-renderer';
import AlertMessage from '../src/components/AlertMessage';

describe('Alert Message Component', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(
            <AlertMessage visible={true} message={['Test message']} onClose={() => {}} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

