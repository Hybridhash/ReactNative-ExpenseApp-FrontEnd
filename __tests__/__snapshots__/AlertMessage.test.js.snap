// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Message Component renders component correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Test message
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#4455BB",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Close
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
